crowbar_framework/app/models/service_object.rb

```diff
@@ -1540,6 +1524,79 @@ def wait_for_chef_daemons(node_list)
 
   private
 
+  THREAD_POOL_SIZE = 20
+
+  def release_chef_locks(locks)
+    return if locks.empty?
+
+    queue = Queue.new
+    locks.each { |l| queue.push l }
+
+    workers = (1...THREAD_POOL_SIZE).map do
```

Comment:
I think it's either:

~~~
irb(main):011:0> (0...3).map { |i| "#{i}" }
=> ["0", "1", "2"]
~~~

or

~~~
irb(main):012:0> (1..3).map { |i| "#{i}" }
=> ["1", "2", "3"]
~~~

where in this examples `THREAD_POOL_SIZE = 3`.